### PR TITLE
WC2-361: Calculations for average weight gain

### PR DESCRIPTION
**Pull request by butofleury**
Explain what problem this PR is resolving

Related JIRA tickets : [WFPPM-16](https://bluesquare.atlassian.net/browse/WFPPM-16)

## Self proofreading checklist

- [x] Did I use eslint and black formatters
- [ ] Is my code clear enough and well documented
- [ ] Are my typescript files well typed
- [ ] New translations have been added or updated if new strings have been introduced in the frontend
- [ ] My migrations file are included
- [ ] Are there enough tests
- [ ] Documentation has been included (for new feature)

## Changes
Added new fields in the Journey table:
   

- initial weight: the weigth at the starting of the journey
-    discharge weight: the weigth on exiting the program
-    start date: the admission date
-    end date: the exit date
-    duration: number of days between admission and exiting
-    weigth loss: the difference between discharge weight and initial weight weight when it's negative    
-    weight_gain: the average of weight gain when the the difference between discharge weight and initial weight weight when is positive

## How to test
From the local instance with wfp coda database, Lunch with docker-compose run iaso manage wfp_etl the ETL script in order to generate data for tableau dashboard and go to the debug page for the beneficiary: e.g: http://localhost:8081/wfp/debug/325/
You should see the calulcated average weight gain. 

e.g: http://localhost:8081/wfp/debug/325/   
          - initial weight = 8 Kgs
          - discharge weight = 10 Kgs
          - start date = 02 Feb 2024
          - end date = 1 March 2024
          - duration = days between start and end date = 28 days
          - weight_gain = (discharge weight - initial weight) / (initial weight x initial weight) ==> (10 - 8) / (8 x 28) = 0.0089 Kgs
          
Sélectionnez-l’objet-journey-à-changer-Iaso
 

## Print screen / video
   - Added fields in Journey table
image

  - Showing new fields in debug page: e.g: http://localhost:8081/wfp/debug/325/
       
   showing-additional-fields-wfp-debug-325

[WFPPM-16]: https://bluesquare.atlassian.net/browse/WFPPM-16?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: plugins/wfp/management/commands/wfp_etl_Under5.py [hunk 1]
```diff
@@ -10,19 +10,38 @@
...
-        return weight_gain
+            initial_weight = float(initial_weight)
+            current_weight = float(current_weight)
+            weight_difference = round((current_weight - initial_weight), 4)
+
+            if weight_difference >= 0:
+                if duration > 0 and current_weight > 0:
+                    weight_gain = round((weight_difference / (initial_weight * float(duration))), 4)
+            elif weight_difference < 0:
+                weight_loss = abs(weight_difference)
```
Comment (madewulf) [inline plugins/wfp/management/commands/wfp_etl_Under5.py:27]: Is it normal that weight_gain seems to be a percentage and weight loss the exact value?

Comment (butofleury) [reply plugins/wfp/management/commands/wfp_etl_Under5.py:27]: yes, the formula to calculate the weight gain has been shared by Sofia. It's an average of weight gained by the beneficiary during the journey(from admission in a program to exit it). The weight loss was not supposed to be calculated, but I proposed to calculate it also with that formula and Sofia was agree!

You can find more details in the last comment in: https://dev.azure.com/worldfoodprogramme/CODA2/_workitems/edit/241569

**Review by madewulf — APPROVED** [review]
